### PR TITLE
abi.py: update _intel_compiler_compare for v0.20

### PR DESCRIPTION
**Pull request by becker33**
Update code in abi.py to match new version interface. This was an oversight in v0.20

@tgamblin this is a PR to fix the abi.py incompatibility Mike mentioned to us.

### CODE REVIEW

File: lib/spack/spack/abi.py [hunk 1]
```diff
@@ -82,9 +82,9 @@ def _intel_compiler_compare(self, pversion, cversion):
         are ABI compatible"""
 
         # Test major and minor versions.  Ignore build version.
-        if len(pversion.version) < 2 or len(cversion.version) < 2:
-            return False
-        return pversion.version[:2] == cversion.version[:2]
+        pv = pversion.lo
+        cv = cversion.lo
```
Comment (alalazo) [inline lib/spack/spack/abi.py:90]: Do we always get `ClosedOpenRange`s from the calling function?

Comment (becker33) [reply lib/spack/spack/abi.py:90]: Yes

Comment (alalazo) [reply lib/spack/spack/abi.py:90]: @becker33 Can we add type-hints to `abi.py` to have a check on that? Otherwise this LGTM

Comment (tldahlgren) [reply lib/spack/spack/abi.py:90]: FWIW.  I tried to help by adding type hints to this method BUT I don't seem to be able to push the changes to this PR.

**Review by alalazo — APPROVED** [review]

**Pull request merged**